src/components/Dialog/index.tsx

```diff
@@ -105,19 +105,36 @@ export function Outer({
     [setOpenIndex, setDialogIsOpen, control.id],
   )
 
+  // This is the function that we call when we want to dismiss the dialog.
   const close = React.useCallback<DialogControlProps['close']>(cb => {
-    if (cb && typeof cb === 'function') {
-      if (closeCallback.current) {
-        logger.error(
-          `Dialog close was passed multiple callbacks, you shouldn't do that`,
-        )
-      }
-      closeCallback.current = cb
+    if (typeof cb === 'function') {
+      closeCallbacks.current?.push(cb)
+    }
+
+    if (!isClosing.current) {
+      sheet.current?.close()
+      isClosing.current = true
     }
-    // initiates a close animation, the actual "close" happens in `onCloseInner`
-    sheet.current?.close()
   }, [])
 
+  // This is the actual thing we are doing once we "confirm" the dialog. We want the dialog's close animation to
+  // happen before we run this. It is passed to the `BottomSheet` component.
+  const onCloseAnimationComplete = React.useCallback(() => {
+    // This removes the dialog from our list of stored dialogs. Not super necessary on iOS, but on Android this
+    // tells us that we need to toggle the accessibility overlay setting
+    setDialogIsOpen(control.id, false)
+    setOpenIndex(-1)
+
+    if (closeCallbacks.current && closeCallbacks.current.length > 0) {
+      for (const cb of closeCallbacks.current) {
+        try {
+          cb()
+        } catch {}
```

Comment:
should not silence them — need to store at least the first one and rethrow at the end of the function